App.tsx

```diff
@@ -17,102 +19,66 @@ import {
-  title: string;
-}>;
 
-function Section({children, title}: SectionProps): React.JSX.Element {
-  const isDarkMode = useColorScheme() === 'dark';
-  return (
-    <View style={styles.sectionContainer}>
-      <Text
-        style={[
-          styles.sectionTitle,
-          {
-            color: isDarkMode ? Colors.white : Colors.black,
-          },
-        ]}>
-        {title}
-      </Text>
-      <Text
-        style={[
-          styles.sectionDescription,
-          {
-            color: isDarkMode ? Colors.light : Colors.dark,
-          },
-        ]}>
-        {children}
-      </Text>
-    </View>
-  );
-}
+import Hea from './src/components/header';
+import Clear from './src/components/clear';
+import Section from './src/components/section';
+
+
 
 function App(): React.JSX.Element {
+
+  interface Element {
+    title: string;
+    description: string;
+    checked: boolean;
+  }
+
+  function selectCallback(isChecked: boolean, elem: Element) {
+    const actualizedTodo = todoS.map(e => {
+      if (e == elem) { e.checked = isChecked; }
+      return e;
+    })
+    setTodos(actualizedTodo);
+  }
+
+  function clearAllDone() {
+    const clearTodos = todoS.filter(obj => {
+      return obj.checked === false;
+    });
+    if (clearTodos.length !== todoS.length) {
+      setTodos(clearTodos);
+    }
+  }
+
   const isDarkMode = useColorScheme() === 'dark';
+  const [todoS, setTodos] = useState([
+    { title: 'prueba', description: 'hasdf', checked: false },
+    { title: 'prueba2', description: 'hasdf', checked: false },
+  ]);
 
   const backgroundStyle = {
-    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
+    flex: 1,
+    backgroundColor: Colors.backGround,
   };
 
   return (
     <SafeAreaView style={backgroundStyle}>
+      <Hea title="Todo"></Hea>
       <StatusBar
         barStyle={isDarkMode ? 'light-content' : 'dark-content'}
-        backgroundColor={backgroundStyle.backgroundColor}
       />
       <ScrollView
         contentInsetAdjustmentBehavior="automatic"
         style={backgroundStyle}>
-        <Header />
-        <View
-          style={{
-            backgroundColor: isDarkMode ? Colors.black : Colors.white,
-          }}>
-          <Section title="Step One">
-            Edit <Text style={styles.highlight}>App.tsx</Text> to change this
-            screen and then come back to see your edits.
-          </Section>
-          <Section title="See Your Changes">
-            <ReloadInstructions />
-          </Section>
-          <Section title="Debug">
-            <DebugInstructions />
-          </Section>
-          <Section title="Learn More">
-            Read the docs to discover what to do next:
-          </Section>
-          <LearnMoreLinks />
-        </View>
+        {todoS.map((todo, index) => (
+          <Section key={index} elem={todo} selectCallback={selectCallback} />
+        ))}
+        <Clear text="clear all done" callback={clearAllDone}></Clear>
```

Comment:
We call `onPress` the "callbacks" / triggers of an interactive component (e.g. a Button). Also, remember to do the components mnemonics so they are easy to understand. What does it mean to have a component named "Clear" in real life?

```suggestion
        <ClearButton text="clear all done" onPress={clearAllDone} />
```

(You can even call `clearAllDone` as `onPressClearAllDone` but is a matter of personal preference)